### PR TITLE
[FIX] add emoji picker to custom status edit modal and file description field

### PR DESCRIPTION
**Pull request by Kunalvrm555**
- [x] have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  - [x] I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - [x] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  - [ ] I have added necessary documentation (if applicable)
  - [x] Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)

https://user-images.githubusercontent.com/67605729/160285146-fabf33ae-5a05-4f1b-bf38-234f018320af.mp4

https://user-images.githubusercontent.com/67605729/161079041-de2181fb-55ac-45e3-98d4-c8a490626f4b.mp4

## Issue(s)

 #24627 #25009
## Steps to test or reproduce

 #24627 #25009

### CODE REVIEW

File: client/sidebar/header/EditStatusModal.tsx [hunk 1]
```diff
@@ -49,6 +49,25 @@ const EditStatusModal = ({ onClose, userStatus, userStatusText }: EditStatusModa
 		onClose();
 	}, [dispatchToastMessage, statusType, statusText, setUserStatus, onClose, t]);
 
+	// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+	const handleEmojiPicker = (event: any) => {
+		const className = document.getElementsByClassName('custome_emoji_react')[0];
+		const inputField = className?.childNodes[0];
```
Comment (nishant23122000) [inline client/sidebar/header/EditStatusModal.tsx:55]: Use useref hook insted of document.getElement..

Comment (Kunalvrm555) [reply client/sidebar/header/EditStatusModal.tsx]: resolved in the next [commit ] in this PR.(https://github.com/RocketChat/Rocket.Chat/pull/24967/commits/19b631ba0396757c9608c91406f49602d0596b98)
Thank you very much @nishant23122000

**Comment by Kunalvrm555** [comment]
@debdutdeb @dudanogueira I have made further changes to add the emoji picker to the file description field as suggested by @nishant23122000. This PR, therefore, solves https://github.com/RocketChat/Rocket.Chat/issues/24627 as well as https://github.com/RocketChat/Rocket.Chat/issues/25009

**Comment by Himanshu664** [comment]
@dougfabris please review this PR it is amazing work

File: apps/meteor/client/sidebar/header/EditStatusModal.tsx [hunk 2]
```diff
@@ -46,6 +46,21 @@ const EditStatusModal = ({ onClose, userStatus, userStatusText }: EditStatusModa
 		onClose();
 	}, [dispatchToastMessage, statusType, statusText, setUserStatus, onClose, t]);
 
+	const inputRef = useRef(null);
+	const handleEmojiPicker = (event: any): void => {
```
Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:50]:
```suggestion
	const handleEmojiPicker = (event: SyntheticEvent): void => {
```

File: apps/meteor/client/sidebar/header/EditStatusModal.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import type { IUser } from '@rocket.chat/core-typings';
 import { Field, TextInput, FieldGroup, Modal, Icon, ButtonGroup, Button } from '@rocket.chat/fuselage';
 import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
 import { useToastMessageDispatch, useSetting, useMethod, useTranslation } from '@rocket.chat/ui-contexts';
-import React, { ReactElement, useState, ChangeEvent, useCallback } from 'react';
+import React, { ReactElement, useState, useRef, ChangeEvent, useCallback } from 'react';
```
Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:4]:
```suggestion
import React, { ReactElement, SyntheticEvent, useState, useRef, ChangeEvent, useCallback } from 'react';
```
```suggestion
import React, { ReactElement, useState, useRef, ChangeEvent, useCallback } from 'react';
```

Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:4]:
```suggestion
import React, { ReactElement, SyntheticEvent, useState, useRef, ChangeEvent, useCallback } from 'react';
```
```suggestion
import React, { ReactElement, useState, useRef, ChangeEvent, useCallback } from 'react';
```

File: apps/meteor/client/sidebar/header/EditStatusModal.tsx [hunk 4]
```diff
@@ -46,6 +46,21 @@ const EditStatusModal = ({ onClose, userStatus, userStatusText }: EditStatusModa
...
+	const handleEmojiPicker = (event: any): void => {
+		event.stopPropagation();
+		event.preventDefault();
+		if (EmojiPicker.isOpened()) {
+			EmojiPicker.close();
+		}
+		EmojiPicker.open(inputRef.current, (emoji: any) => {
+			const emojiValue = `:${emoji}:`;
+			inputRef.current.focus();
+			if (!document.execCommand || !document.execCommand('insertText', false, emojiValue)) {
```
Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:59]: `execCommand` is deprecated. Don't know enough about the EmojiPicker code at this moment so not going to advice anything.

Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:59]: `execCommand` is deprecated. Don't know enough about the EmojiPicker code at this moment so not going to advice anything.

File: apps/meteor/client/sidebar/header/EditStatusModal.tsx [hunk 5]
```diff
@@ -46,6 +46,21 @@ const EditStatusModal = ({ onClose, userStatus, userStatusText }: EditStatusModa
 		onClose();
 	}, [dispatchToastMessage, statusType, statusText, setUserStatus, onClose, t]);
 
+	const inputRef = useRef(null);
```
Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:49]:
```suggestion
	const inputRef = useRef<HTMLInputElement>(null);
```

Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:49]:
```suggestion
	const inputRef = useRef<HTMLInputElement>(null);
```

File: apps/meteor/client/sidebar/header/EditStatusModal.tsx [hunk 6]
```diff
@@ -46,6 +46,21 @@ const EditStatusModal = ({ onClose, userStatus, userStatusText }: EditStatusModa
...
+	const handleEmojiPicker = (event: any): void => {
+		event.stopPropagation();
+		event.preventDefault();
+		if (EmojiPicker.isOpened()) {
+			EmojiPicker.close();
+		}
+		EmojiPicker.open(inputRef.current, (emoji: any) => {
```
Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:56]:
```suggestion
		EmojiPicker.open(inputRef.current, (emoji: string) => {
```

Comment (debdutdeb) [inline apps/meteor/client/sidebar/header/EditStatusModal.tsx:56]:
```suggestion
		EmojiPicker.open(inputRef.current, (emoji: string) => {
```
```suggestion
		EmojiPicker.open(inputRef.current, (emoji: any) => {
```

**Review by debdutdeb — CHANGES_REQUESTED** [review]
Same for the other file. 

That said, I don't think this is the right approach. I have something in mind, can't share just yet. For starters there's the same code now in three different places, big no-no. 

@dudanogueira hey, can you add this to your list for our discussion next week? would be super helpful! thanks 🥸

**Comment by Kunalvrm555** [comment]
@debdutdeb can we make any progress in this feature yet? Did you find any workaround for this ? Please let me know

**Comment by successgilli** [comment]
Can I add a commit to this?

**Pull request closed without merging**